### ISSUE TITLE
Prettifying files in place in Gulp 4

### ISSUE THREAD

**Issue by PyvesB** [bhargavrpatel/gulp-prettier#15]
Hello,

I want to run Prettier on my JavaScript files in place, and following advice from https://github.com/bhargavrpatel/gulp-prettier/issues/5#issuecomment-385441102, I've set up a _gulpfile.js_ file similar to the following:
```
const gulp = require("gulp");
const prettier = require("gulp-prettier");

gulp.task("prettier", () =>
  gulp.src("*.js")
    .pipe(prettier({ singleQuote: true }))
      .pipe(gulp.dest(file => file.base))
);
```
The _package-lock.json_ is:
```
{
    "devDependencies": {
        "gulp": "^4.0.2",
        "gulp-prettier": "^2.1.0"
    }
}
```
However, this does not seem to work as expected unless I downgrade the Gulp version to 3.9.1.

How would I go about prettifying files in place using Gulp 4+?

Thanks in advance,

Pyves

**Comment by TheDancingCode** [comment]
Hi Pyves,

What's not working? Are you getting an error?

**Comment by PyvesB** [comment]
No errors, nothing seems to happen. With Gulp 4.0.2, _gulpfile.js_ is unchanged after running the prettier task, but with Gulp 3.9.1, the file is prettified as expected.

**Comment by TheDancingCode** [comment]
It works just fine over here, so I'm not sure what the issue is. Having said that, the more idiomatic way of writing this task in gulp 4 would be:
```js
const { src, dest } = require('gulp');
const prettier = require('gulp-prettier');

function prettify() {
  return src('*.js')
    .pipe(prettier({ singleQuote: true }))
    .pipe(dest(file => file.base));
}

exports.default = prettify;
```

**Comment by PyvesB** [comment]
I'm using macOS Mojave 10.14.5.

I've got some interesting observations. It does actually work, however my IDE (Eclipse) wasn't picking up the file changes unless I opened and closed the file again. This worked fine in Gulp 3.9.1, so something has changed with file handling in Gulp 4 which causes the IDE to no longer detect any changes. Other users have reported similar issues with the `watch` function no longer monitoring file changes properly (for example: https://github.com/gulpjs/gulp/issues/2293), my guess would be that it's somehow related.

Note that I've tried the same IDE on a Windows 10 machine, no issue to report in either Gulp 3 or Gulp 4.

**Comment by PyvesB** [comment]
After a bit of digging, I noticed that in Gulp 4 the "last modified" file attribute is not updated after prettifying the files, whereas it is in Gulp 3. This will break any editors that rely on this attribute for detecting file changes.

This is apparently a known issue that has been around for a few years now (see https://github.com/gulpjs/gulp/issues/1461 and https://github.com/gulpjs/vinyl/issues/105#issuecomment-435705083).

Bottom line is: nothing wrong with gulp-prettier, but there is an issue in Gulp's file metadata handling. Therefore, I'll close this issue, thank you for your time!
